testsuite/cluster-test/src/experiments/twin_validator.rs

```diff
@@ -0,0 +1,152 @@
+        let mut instances = cluster.validator_instances().to_vec();
+        let mut twin_validators = vec![];
+        let mut rnd = rand::thread_rng();
+        for _i in 0..self.pair {
+            twin_validators.push(instances.remove(rnd.gen_range(0, instances.len())));
+        }
+        Self::E {
+            instances,
+            twin_validators,
+        }
+    }
+}
+
+#[async_trait]
+impl Experiment for TwinValidators {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&self.twin_validators)
+    }
+
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        let buffer = Duration::from_secs(60);
+        let window = Duration::from_secs(240);
+        let mut new_instances = vec![];
+        for i in 0..self.twin_validators.len() {
+            let mut new_twin_config = self.twin_validators[i].instance_config().clone();
+            new_twin_config.make_twin(1);
+            let new_inst = context
+                .cluster_swarm
+                .spawn_new_instance(new_twin_config, true)
+                .await?;
+            new_instances.push(new_inst.clone());
+        }
+        let instances = self.instances.clone();
+        let emit_job_request =
+            EmitJobRequest::for_instances(instances, context.global_emit_job_request);
+        let emit_txn = context.tx_emitter.emit_txn_for(window, emit_job_request);
+        let trace_tail = &context.trace_tail;
+        let trace_delay = buffer;
+        let trace = false;
+        let capture_trace = async move {
+            if trace {
+                tokio::time::delay_for(trace_delay).await;
+                Some(trace_tail.capture_trace(Duration::from_secs(5)).await)
+            } else {
+                None
+            }
+        };
+        let (stats, _trace) = join!(emit_txn, capture_trace);
+        let stats = stats?;
+        let end = unix_timestamp_now() - buffer;
+        let start = end - window + 2 * buffer;
+        let avg_txns_per_block = stats::avg_txns_per_block(&context.prometheus, start, end);
+        let avg_txns_per_block = avg_txns_per_block
+            .map_err(|e| warn!("Failed to query avg_txns_per_block: {}", e))
+            .ok();
+        let avg_latency_client = stats.latency / stats.committed;
+        let p99_latency = stats.latency_buckets.percentile(99, 100);
+        let avg_tps = stats.committed / window.as_secs();
+        info!(
+            "Link to dashboard : {}",
+            context.prometheus.link_to_dashboard(start, end)
+        );
+        info!(
+            "Tx status: txn {}, avg latency {}",
+            stats.committed as u64, avg_latency_client
+        );
+        let submitted_txn = stats.submitted;
+        let expired_txn = stats.expired;
+        let futures: Vec<_> = new_instances.iter().map(|ic| ic.stop()).collect();
+        try_join_all(futures).await?;
+        time::delay_for(Duration::from_secs(10)).await;
+        context
```

Comment:
A lot of code here related to metric reporting seem to be just a copy paste from perf experiment.
That's not great, should we add a method in `SuiteReport` like `report_txn_stats(stats: TxStats, window)`?